File: frontend/app/dtos/MuseumObjectDTO.ts
import type { NoteDTO } from "./NoteDTO";
import type { ImageDTO } from "./ImageDTO";

export interface MuseumObjectDTO {
    id: number;
    objectName: string;
    groupName: string;
    technicalData: string;
    description: string;
    type: string;
    category: string;
    notes: NoteDTO[];
    isSeen: boolean;
    image: ImageDTO;
}

File: backend/src/main/java/es/codeurjc/daw/museum/dto/MuseumObjectDTO.java
package es.codeurjc.daw.museum.dto;

import java.util.List;

public record MuseumObjectDTO(
    Long id,
    String objectName,
    String groupName,
    String technicalData,
    String description,
    String type,
    String category,
    List <NoteBasicDTO> notes,
    boolean isSeen,
    ImageDTO image
) {}
